File: lab_2/lab_5/src/kustova/natalia/AbstractFileSystemNode.java
package kustova.natalia;

public class AbstractFileSystemNode implements FileSystemNode{
    private FileSystemNode parent;
    private String name;

    public void setParent(FileSystemNode parent) {
        this.parent = parent;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public final FileSystemNode getParentElement() {
        return parent;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public final String getPath() {
        if(parent == null) return "://" + name;
        return parent.getPath() + "/" + name;
    }
}

File: lab_3/lab_5/src/main/kotlin/AbstractFileSystemNode.kt
public open class AbstractFileSystemNode public constructor() : kustova.natalia.FileSystemNode {
    private final var name: kotlin.String?

    private final var parent: kustova.natalia.FileSystemNode?

    public open fun getName(): kotlin.String

    public open fun getParentElement(): kustova.natalia.FileSystemNode

    public open fun getPath(): kotlin.String

    public final fun setName(name: kotlin.String?): kotlin.Unit

    public final fun setParent(parent: kustova.natalia.FileSystemNode?): kotlin.Unit
}